marketplace/deployer_helm_tiller_base/bin/deploy_internal.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+#
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eox pipefail
+
+app_uid=$(kubectl get "applications/$NAME" \
+  --namespace="$NAMESPACE" \
+  --output=jsonpath='{.metadata.uid}')
+app_api_version=$(kubectl get "applications/$NAME" \
+  --namespace="$NAMESPACE" \
+  --output=jsonpath='{.apiVersion}')
+
+for chart in /data/chart/*; do
+  # Expand and apply the template for the Application resource.
+  # Note: This must be done out-of-band because the Application resource
+  # is created before the deployer is invoked, but helm does not handle
+  # pre-existing resources.
+  helm template \
+      --name="$NAME" \
+      --namespace="$NAMESPACE" \
+      --values=<(print_config.py --output=yaml) \
+      --set "$(cat /data/schema.yaml \
+                 | yaml2json \
+                 | jq -r '.properties
+                            | to_entries
+                            | .[]
+                            | select(.value["x-google-marketplace"].type == "APPLICATION_UID")
+                            | .key')=" \
+      "$chart" \
+    | yaml2json \
+    | jq 'select( .kind == "Application" )' \
```

Comment:
(perhaps a follow up): Set proper labels for the application

Reply:
I thought that application labels ought to be set by `scripts/install` (or our backend)?

Reply:
But it would be overwritten by the one from the chart. I'm sure about the behavior of the current deployer. Likely that it doesn't alter the `Application` manifest. We don't have to do it here, but it's a good consistent thing to do, since we're already modifying the labels of the helm resources.

Reply:
Makes sense and agreed. Will take as a follow up.